### PR TITLE
[5/5 verity read-only root] ISO support, license and check cleanups

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- ~~[ ] Documentation has been updated to match any changes to the build system~~ Documentation to follow later.
- [ ] Ready to merge

---

###### Summary 

Read-only root and dm-verity: This set of 5 PRs will enable dm-verity based read-only root file systems which can detect, and optionally repair modifications to files on the root file system. This supports both basic resilience to corruption as well as a number of additional security features.

PR [5/5] Supports the ISO installer tool, and finishes some cleanup and book keeping.

###### Change Log  

- Attended installer will now work with these PRs
  - Fix the kernel packages for the ISOs, (we were still including them via package list rather than config)
  - Set the `dmroot` flag for the device mapper root when using encryption (We can no name the root drives anything we want)
  - Copy in any additional kernel command line settings defined in the image config when using ISO flow
  - ** The ISO installer is able to create and measure dm-verity disks quite well, however there is a bug with the initramfs regeneration package which ends up creating an almost completely empty initramfs. Need to fix that before dm-verity support from the ISO installer works**
- Update license and check related files.

###### Does this affect the toolchain?  

NO

###### Test Methodology
Local builds and testing via hyper-v

### CODE REVIEW

File: toolkit/tools/imagegen/configuration/configuration_test.go [hunk 1]
```diff
@@ -91,8 +91,8 @@ func TestShouldFailDeviceMapperWithNoRootDisks(t *testing.T) {
 	var checkedConfig Config
 	testConfig := expectedConfiguration
 
-	// Clear the disks
-	testConfig.Disks = []Disk{}
+	// Clear the disks, and one empty one
```
Comment (MateuszMalisz) [inline toolkit/tools/imagegen/configuration/configuration_test.go:94]: I think you meant "add empty one".

File: toolkit/tools/imagegen/diskutils/initramfs.go [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+			err = nil
+			break
+		}
+		if err != nil {
+			return
+		}
+
+		// For a given symlink, generate a payload that contains the read value of the link.
+		// The payload should be written after the header.
+		// e.g. A link from (/bin -> /usr/bin) would have a payload of "/usr/bin").
```
Comment (schmittjoseph) [inline toolkit/tools/imagegen/diskutils/initramfs.go:86]: extra `)` at end of comment.

**Review by schmittjoseph — APPROVED** [review]

File: toolkit/tools/imagegen/attendedinstaller/_manualrun/manualrun.go [hunk 3]
```diff
@@ -23,13 +23,22 @@ func main() {
...
 				Name: "Core",
 				PackageLists: []string{
-					"core-packages-image.json",
-					"hyperv.json",
+					"packagelists/core-packages-image.json",
+					"packagelists/hyperv-packages.json",
+				},
+				KernelOptions: map[string]string{
+					"default": "kernel",
+					"hyperv":  "kernel-hyperv",
```
Comment (christopherco) [inline toolkit/tools/imagegen/attendedinstaller/_manualrun/manualrun.go:41]: Do not include the hyperv kernel option. It is not ready for use.

File: toolkit/tools/imagegen/attendedinstaller/_manualrun/manualrun.go [hunk 4]
```diff
@@ -52,10 +61,40 @@ func main() {
 					"developer-packages.json",
 					"hyperv.json",
 				},
+				KernelOptions: map[string]string{
+					"default": "kernel",
+					"hyperv":  "kernel-hyperv",
```
Comment (christopherco) [inline toolkit/tools/imagegen/attendedinstaller/_manualrun/manualrun.go:66]: Do not include the hyperv kernel option. It is not ready for use.

File: toolkit/tools/imagegen/attendedinstaller/_manualrun/manualrun.go [hunk 5]
```diff
@@ -52,10 +61,40 @@ func main() {
...
+			configuration.SystemConfig{
+				Name: "Read-Only",
+				PackageLists: []string{
+					"packagelists/core-packages-image.json",
+					"packagelists/hyperv-packages.json",
+					"packagelists/read-only-root-packages.json",
+				},
+				KernelOptions: map[string]string{
+					"default": "kernel",
+					"hyperv":  "kernel-hyperv",
```
Comment (christopherco) [inline toolkit/tools/imagegen/attendedinstaller/_manualrun/manualrun.go:91]: Do not include the hyperv kernel option. It is not ready for use.

**Review by christopherco — APPROVED** [review]

**Pull request merged**